domain/service/servicedetails_store.go

```diff
@@ -73,6 +62,32 @@ func (s *storeImpl) GetAllServiceDetails(ctx datastore.Context, since time.Durat
 		}
 
 		parentMap[d.ParentServiceID] = struct{}{}
+
+		if r != nil && !r.MatchString(d.Name) {
+			continue
+		}
+
+		if query.Since > 0 && d.UpdatedAt.Before(since) {
```

Comment:
Why not include the since in the elastic query if present? Would reduce the number of results.